### ISSUE TITLE
unsetting non required fields still validates them

### ISSUE THREAD

**Issue by despairblue** [Automattic/mongoose#3217]
```javascript
const mongoose = require('mongoose')

mongoose.connect('mongodb://localhost:27017', function () {
  const userSchema = new mongoose.Schema({
    username: {
      type: String,
      required: true
    },
    email: {
      type: String,
      validate: function (email) {
        return /.*@.*/.test(email)
      }
    }
  })

  const userModel = mongoose.model('User', userSchema)

  userModel.create({
    username: 'river'
  }, function (error, user) {
    if (error) {
      console.log(error)
    } else {
      console.log('creating w/o email worked')
      user.email = 'river@firefly.space'

      user.save(function (error, user) {
        if (error) {
          console.log(error)
        } else {
          console.log('saving with email worked')
          user.email = undefined

          user.save(function (error) {
            if (error) {
              console.log('saving w/o email did not work')
              console.log(error)
              process.exit(1)
            } else {
              process.exit(0)
            }
          })
        }
      })
    }
  })
})
```

Output:
```
creating w/o email worked
saving with email worked
saving w/o email did not work
{ [ValidationError: Validation failed]
  message: 'Validation failed',
  name: 'ValidationError',
  errors: 
   { email: 
      { [ValidatorError: Validator failed for path `email` with value ``]
        message: 'Validator failed for path `email` with value ``',
        name: 'ValidatorError',
        path: 'email',
        type: 'user defined',
        value: undefined } } }
```

This strikes me as not really consistent. Is this intended behavior?

**Comment by vkarpov15** [comment]
Can you confirm which version of mongoose?

**Comment by despairblue** [comment]
3.8.25

I could try to update to 3.8.34.

**Comment by vkarpov15** [comment]
This behavior was fixed for 4.0.x, see #1594. However, this will not be backported to 3.8 as it's potentially backwards breaking.

**Comment by despairblue** [comment]
ah, good to know :bulb:

thx :bow:
